### ISSUE TITLE
Can't build 2.12 from source tarball: "fatal: Not a git repository"

### ISSUE THREAD

**Issue by SethTisue** [scala/scala-dev#220]
if you don't have a `.git` directory at the root, then: 

```
fatal: Not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
```

and then

```
java.lang.StringIndexOutOfBoundsException: String index out of range: 7
    at java.lang.String.substring(String.java:1963)
    at VersionUtil$$anonfun$versionPropertiesImpl$1.apply(VersionUtil.scala:81)
    at VersionUtil$$anonfun$versionPropertiesImpl$1.apply(VersionUtil.scala:61)
    at sbt.EvaluateSettings$MixedNode.evaluate0(INode.scala:175)
```

reported by Jörg Plate at https://groups.google.com/forum/#!topic/scala-user/T9KHTQDtUbI . I can recall this coming up before, but I don't remember where.

at the time I think we thought something like "well that's unfortunate, but it isn't likely to come up much"

**Comment by SethTisue** [comment]
as a workaround you can do:

```
git init
git commit -m "" --allow-empty --allow-empty-message
```

before running sbt

**Comment by Patterner** [comment]
It works, but I hope new tarballs won't need that workaround.

**Comment by gourlaysama** [comment]
I vaguely recalled this problem from somewhere, so I just went and fixed it: scala/scala#5411.

**Comment by SethTisue** [comment]
LGTM

**Comment by gourlaysama** [comment]
Fixed in scala/scala#5411. Thanks @Patterner for reporting! :smile:
